index.bs

```diff
@@ -371,13 +389,113 @@ model]]:
+        [=top-level traversable/bounce tracking record=] is not null.
+    1. [=set/Append=] |origin|'s [=host=] to |topLevelTraversable|'s
+        [=top-level traversable/bounce tracking record=]'s
+        [=bounce tracking record/storage access set=].
+
+Issue: TODO: Handle subresource requests and iframes for client-side redirects.
+
+</div>
+
+<h5 id="bounce-tracking-mitigations-network-cookie-write-monkey-patch">Network
+Cookie Write Monkey Patch</h5>
+
+Insert the following steps in the <a spec="fetch">HTTP-network fetch</a>
+algorithm after step 15 where cookies are parsed and saved in the cookie store:
+
+1. If cookies were stored in the cookie store in the previous step, then
+    run [=process a network cookie access for bounce tracking mitigations=]
+    given <var ignore>request</var>.
+
+<h5 id="bounce-tracking-mitigations-network-cookie-read-monkey-patch">Network
+Cookie Read Monkey Patch</h5>
+
+Insert the following steps in the <a spec="fetch">HTTP-network-or-cache fetch</a>
+algorithm after step 8.21.1.2 where cookies are added to the request headers:
+
+1. If cookies were attached to the request in the previous step, then run
+    [=process a network cookie access for bounce tracking mitigations=] given
+    <var ignore>httpRequest</var>.
+
+<h5 id="bounce-tracking-mitigations-nav-start-monkey-patch">Start Navigation
+Monkey Patch</h5>
+
+<div algorithm>
+
+To <dfn>process navigation start for bounce tracking</dfn> given a [=navigable=]
+|navigable| and [=Document=] |sourceDocument|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. Set |initialHost| to |sourceDocument|'s [=Document/origin=]'s [=host=] if
+    |sourceDocument|'s [=Document/origin=] is not an [=opaque origin=], and
+    empty string otherwise.
+1. Set |navigable|'s [=top-level traversable/bounce tracking record=] to a new
+    [=bounce tracking record=] with [=bounce tracking record/initial host=]
+    set to |initialHost|.
+
+Issue: TODO: Handle programmatic navigations within a particular time frame as
+    client-side redirects.
+
+</div>
+
+Insert the following steps in the <a spec="html">navigate</a> algorithm before
+step 8 where navigation to fragments is handled.
+
+1. Run [=process navigation start for bounce tracking=] given
+    <var ignore>navigable</var> and <var ignore>sourceDocument</var>.
+
+<h5 id="bounce-tracking-mitigations-nav-end-monkey-patch">End Navigation Monkey
+Patch</h5>
+
+<div algorithm>
+
+To <dfn>process navigation end for bounce tracking</dfn> given a [=navigable=]
+|navigable| and [=response/url list=] |URLs|, perform the following steps:
+
+1. If |navigable| is not a [=top-level traversable=], then abort these steps.
+1. If |URLs| is empty, then abort these steps.
+1. Set |initialHost| to |navigable|'s [=top-level traversable/bounce tracking record=]'s
+    [=bounce tracking record/initial host=].
+1. Set |finalHost| to the [=host=] of the last entry in |URLs|.
```

Comment:
I think we need to actually get the etld+1 site from the `[=host=]`.  This was something I had not gotten to in the PR and hadn't made a note about.

To do this I think you can use `[=obtain a site=]` on a non-opaque origin:

https://html.spec.whatwg.org/#obtain-a-site

and then use the host from here:

https://html.spec.whatwg.org/#concept-scheme-and-host-host

We should double check we are using etld+1 site for hosts everywhere in this PR.

Reply:
Done, added calls to [=obtain a site=] under cookie writes, start nav, and end nav.